### ISSUE TITLE
秒记

### ISSUE THREAD

**Issue by zhaipro** [zhaipro/unet#1]
目标分割

**Comment by zhaipro** [comment]
https://blog.csdn.net/trent1985/article/details/80944942
https://gitee.com/onlysea/modihair/
https://keras.io/zh/layers/core/#permute
http://vis-www.cs.umass.edu/lfw/
http://vis-www.cs.umass.edu/lfw/part_labels/
https://www.matrix.com/virtual-hair-color-try-on
https://challenger.ai/dataset/lad2018
https://kazzkiq.github.io/svg-color-filter/
https://docs.rainmeter.net/tips/colormatrix-guide/
https://stackoverflow.com/questions/22999020/how-to-apply-svg-filter-to-group-using-svgwrite-with-python/23291406
https://yanweifu.github.io/papers/hairstyle_v_14_weidong.pdf
https://github.com/lemondan/HumanParsing-Dataset
https://github.com/TianzhongSong/Person-Segmentation-Keras
https://github.com/bonlime/keras-deeplab-v3-plus
http://www-scf.usc.edu/~liwenhu/SHM/database.html
http://mmlab.ie.cuhk.edu.hk/projects/CelebA.html
https://github.com/tensorflow/models/tree/master/research/deeplab
明度，亮度，HSL，HSV
https://hanks.pub/2016/03/26/color-board/
https://open.mtlab.meitu.com/algorithm/imageProcessing/makeup

**Comment by zhaipro** [comment]
1. 没有数据增强
    -  修改色相饱和度
    -  修改尺寸
    -  水平翻转
1. 没有原始训练集
1. 没有尝试更多模型
1. 没有提供多种尺寸的图片用于训练，我想提升计算效率，达到在手机端能实时美发。

**Comment by zhaipro** [comment]
0.12, 0.93, 3.47M
700/700 [=] - 90s 129ms/step - loss: 0.0102 - dice_coeff: 0.9709 - psnr: 28.0166 - val_loss: 0.0110 - val_dice_coeff: 0.9679 - val_psnr: 27.5066

**Comment by zhaipro** [comment]
https://docs.python.org/3/library/zipfile.html
https://stackoverflow.com/questions/18231135/load-compressed-data-npz-from-file-using-numpy-load

```
im = np.frombuffer(fp.read(), dtype='uint8')
im = cv2.imdecode(im, cv2.IMREAD_COLOR)
```

**Comment by zhaipro** [comment]
重要的灵感来源

[Kaggle-Carvana-Image-Masking-Challenge](https://github.com/petrosgk/Kaggle-Carvana-Image-Masking-Challenge)
[头发分割](http://open.mtlab.meitu.com/doc?id=16)
[曝光算法](https://blog.csdn.net/u011043491/article/details/51545466)
[HSL和HSV色彩空间](https://zh.m.wikipedia.org/zh-cn/HSL和HSV色彩空间)

重要的失败经历

LUT
样条插值
[开源的训练集可能还不如收费的预测集](http://www.cs.ubbcluj.ro/~dadi/face-hair-segm-database.html)

不能再忘的基础知识

[幂函数](https://zh.m.wikipedia.org/zh-hans/幂函数)

**Comment by zhaipro** [comment]
距离美发还有很远的路要走，每天路边美女的发型告诉我。

**Comment by zhaipro** [comment]
[在线展示](http://shell.teachx.cn:8086)

**Comment by zhaipro** [comment]
原本计划用染发效果来评价分割效果，但染发的挑战性超出了我的想象。

**Comment by zhaipro** [comment]
`max(color) > 0`
`sum(mask) > 0`

**Comment by zhaipro** [comment]
我的目标
<div>
<img src="https://user-images.githubusercontent.com/8620842/60262438-be22f800-9910-11e9-840c-7b779091bfbf.jpg" height="400">
<img src="https://user-images.githubusercontent.com/8620842/60262446-c11de880-9910-11e9-9fe3-faf252e3f2ee.jpg" height="400">
</div>

**Comment by zhaipro** [comment]
`mask = cv2.GaussianBlur(mask, (5, 5), 5)`

**Comment by zhaipro** [comment]
我回来啦

**Comment by zhaipro** [comment]
从原图中提取明度，并做相应调整，结合上色的色相和饱和度。
现在尝试从原图中提取亮度，并做相应调整，保留原图的饱和度，结合上色的色相。
我们只是染发，修改的越少越好。

我们染发当然换了颜色，也提升了反光强度，让美发更明显，但饱和度是个什么鬼？

**Comment by zhaipro** [comment]
<div>
<img src="https://user-images.githubusercontent.com/8620842/64858225-636c8300-d659-11e9-97c1-3dd4a53972b2.jpg" height="300">
<img src="https://user-images.githubusercontent.com/8620842/64858229-649db000-d659-11e9-812e-d3f25106e8a8.jpg" height="300">
</div>

**Comment by zhaipro** [comment]
是因为我们头发亮度不充分的照片信息量太少了吗？
要不要单独训练一个模型，用于高清化？

**Comment by zhaipro** [comment]
[美图](https://open.mtlab.meitu.com/algorithm/imageProcessing/makeup)给我的染发效果
<div>
<img src="https://user-images.githubusercontent.com/8620842/65397510-45361e00-dde3-11e9-8c7b-47fa3f8170ca.jpg" height="300">
<img src="https://user-images.githubusercontent.com/8620842/65397511-45ceb480-dde3-11e9-8459-e879f6cc29c1.jpg" height="300">
</div>
